com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/transaction/ZigBeeTransactionFuture.java

```diff
@@ -35,6 +42,17 @@
     // Not final for tests
     private static long TIMEOUT_MINUTES = 5;
 
+    /**
+     * Constructur
+     *
+     * @param callback the {@link ZigBeeTransactionCallback} to call on failure
+     * @param transaction the {@link ZigBeTransaction} linked to this
+     */
+    public ZigBeeTransactionFuture(ZigBeeTransactionCallback callback, ZigBeeTransaction transaction) {
```

Comment:
I am wondering whether the `callback` is necessary or if the `transaction` is sufficient, because the `transcation` has a reference to the `transactionManager` through which the transaction was sent. So I think it should be possible to cancel the transaction through this path, instead of adding an additional one which also increases the complexity.

Am I overlooking something here?

Reply:
Thanks. I might have missed this as it's not the `ZigBeeTransaction` that we're trying to cancel, but the `ZigBeeTransactionFuture` so I was focused on getting a link from `ZigBeeTransactionFuture` to the `transaction manager` (or the `queue` as is currently implemented here).

This is a little convoluted whichever way it goes, but I appreciate the feedback and I'll have another look to see if your suggestion looks better.